### ISSUE TITLE
Proper installation of the kernels 

### ISSUE THREAD

**Issue by goatchurchprime** [adafruit/jupyter_micropython_kernel#2]
It's quite a hack to get the kernel.json file into the correct place.  I'm sure there's a proper way of doing this to make it easier. (I don't know it either)

In Linux by running the command:
    jupyter kernelspec list

I found that my kernels were in fact at:
    ~/anaconda3/share/jupyter/kernels/
instead of:
    ~/.local/share/jupyter/kernels

**Comment by goatchurchprime** [comment]
I've ported the install.py method across from the bash_kernel like so:

https://github.com/goatchurchprime/jupyter_micropython_kernel/blob/master/jupyter_micropython_kernel/install.py

(I'm not really up on these labels and pull requests, me)

**Comment by tannewt** [comment]
Hi @goatchurchprime, thanks for you interest in the CircuitPython Jupyter kernel. There is a second attempt at it [here](https://github.com/willingc/circuitpython_kernel) that you may be able to pull some of the things you want from. They were done in parallel so I'd love to hear your take on which one we should improve going forwards.

**Comment by goatchurchprime** [comment]
It's good to know I'm not totally replicating something already done.  (Although it would be better if it was already done).  My bet is that this way could easily become the primary interface to micropython.  It just needs a means of sending some system commands and stuff to connect or check how the connection is behaving.

To do this we need a slightly more sophisticated version of pyboard.py that works in a different process and handles the interface with the serial line (or websocket) cleanly.  Then the main kernel code can be copied from the bash_kernel.   

I've got some almost working experiment here:
   https://github.com/goatchurchprime/jupyter_micropython_kernel/blob/master/asyncmodule.py

The circuitpython one uses a very trivial wrapper of the serial interface, like so:
    https://github.com/willingc/circuitpython_kernel/blob/master/circuitpython_kernel/board.py

As to the feature of transferring files, it seems there's no (or it's unknown) protocol for doing it over the serial line.  Normally you do it over the webrepl on the wifi network.  One solution for doing it on the serial line is done by this project, where it deposits a script into the device that receives the bytes and saves them into a file:   https://github.com/BetaRavener/uPyLoader/tree/master/mcu   I'm hoping there's a better way.

**Comment by goatchurchprime** [comment]
It's good to know I'm not totally replicating something already done.  (Although it would be better if it was already done).  My bet is that this way could easily become the primary interface to micropython.  It just needs a means of sending some system commands and stuff to connect or check how the connection is behaving.

To do this we need a slightly more sophisticated version of pyboard.py that works in a different process and handles the interface with the serial line (or websocket) cleanly.  Then the main kernel code can be copied from the bash_kernel.   

I've got some almost working experiment here:
   https://github.com/goatchurchprime/jupyter_micropython_kernel/blob/master/asyncmodule.py

The circuitpython one uses a very trivial wrapper of the serial interface, like so:
    https://github.com/willingc/circuitpython_kernel/blob/master/circuitpython_kernel/board.py

As to the feature of transferring files, it seems there's no (or it's unknown) protocol for doing it over the serial line.  Normally you do it over the webrepl on the wifi network.  One solution for doing it on the serial line is done by this project, where it deposits a script into the device that receives the bytes and saves them into a file:   https://github.com/BetaRavener/uPyLoader/tree/master/mcu   I'm hoping there's a better way.

**Comment by goatchurchprime** [comment]
@tannewt I think I've knocked out most of the features over at this version -- once I got my teeth into it
    https://github.com/goatchurchprime/jupyter_micropython_kernel

As it's hard to port across the issues, I'm going to close the issues here that are done

**Comment by tannewt** [comment]
Works for me!

**Comment by goatchurchprime** [comment]
Done in https://github.com/goatchurchprime/jupyter_micropython_kernel

More to the point, there's not many of the original lines remaining.  And the whole thing now is about 250 lines.  It'll get bigger if it has to extend to other platforms, but it's nearly at a stage where I can prove it's productive.
